test/e2e/storage/csi_objects.go

```diff
@@ -413,3 +471,185 @@ func deployGCEPDCSIDriver(
 	framework.ExpectNoError(err, "Failed to create DaemonSet: %v", nodeds.Name)
 
 }
+
+func shredFile(filePath string) {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		framework.Logf("File %v was not found, skipping shredding", filePath)
+		return
+	}
+	framework.Logf("Shredding file %v", filePath)
+	_, _, err := framework.RunCmd("shred", "--remove", filePath)
+	if err != nil {
+		framework.Logf("Failed to shred file %v: %v", filePath, err)
+	}
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		framework.Logf("File %v successfully shredded", filePath)
+		return
+	}
+	// Shred failed Try to remove the file for good meausure
+	err = os.Remove(filePath)
+	framework.ExpectNoError(err, "Failed to remove service account file %s", filePath)
+
+}
+
+// createGCESecrets downloads the GCP IAM Key for the default compute service account
+// and puts it in a secret for the GCE PD CSI Driver to consume
+func gceSecrets(client clientset.Interface, config framework.VolumeTestConfig) *serviceAccountKey {
+	saFile := "/tmp/cloud-sa.json"
+	var revokeSAKey bool
+	if premadeSAFile, ok := os.LookupEnv("E2E_GOOGLE_APPLICATION_CREDENTIALS"); ok {
+		framework.Logf("Found CI service account key at %v", premadeSAFile)
+		// Need to copy it saFile
+		stdout, stderr, err := framework.RunCmd("cp", premadeSAFile, saFile)
+		framework.ExpectNoError(err, "error copying service account key: %s\nstdout: %s\nstderr: %s", err, stdout, stderr)
+		defer shredFile(saFile)
+		revokeSAKey = false
+	} else {
+		// Could not find pre-made CI Key so we can create our own
```

Comment:
do we still need to have this auto generation logic?  If you're running it manually, we could also require you setup the key separately using the existing scripts.

Reply:
hmm yes, we could but then the test wouldn't run properly automatically. It is definitely a tradeoff. I prefer it to be fully automated like this but I could also be convinced the other way

Reply:
consider the case where someone just wants to run all storage tests, its weird to just have the CSI GCE ones fail because the user wasn't present when prompted to do some manual actions.

Reply:
Can we just require that the env variable must be set to run the gce pd tests?  It could be a bit alarming to have the test unexpectently create and download keys from the project.  That means the user also has to run the e2e tests with a gcp account that has a lot of permissions

Reply:
good point, i'll have it spit out a message to set the environment variable